Index: webrtc/sources/system_wrappers/source/Android.mk
===================================================================
--- webrtc.orig/sources/system_wrappers/source/Android.mk	2012-09-26 23:03:45.086734939 +0200
+++ webrtc/sources/system_wrappers/source/Android.mk	2012-09-26 23:03:47.814734825 +0200
@@ -20,6 +20,7 @@
     android/cpu-features.c \
     cpu_features_android.c \
     map.cc \
+    rw_lock_generic.cc \
     sort.cc \
     aligned_malloc.cc \
     atomic32_posix.cc \
Index: webrtc/sources/system_wrappers/source/rw_lock.cc
===================================================================
--- webrtc.orig/sources/system_wrappers/source/rw_lock.cc	2012-09-26 23:03:45.114734938 +0200
+++ webrtc/sources/system_wrappers/source/rw_lock.cc	2012-09-26 23:04:04.238734137 +0200
@@ -15,6 +15,8 @@
 #if defined(_WIN32)
 #include "rw_lock_generic.h"
 #include "rw_lock_win.h"
+#elif defined(WEBRTC_ANDROID)
+#include "rw_lock_generic.h"
 #else
 #include "rw_lock_posix.h"
 #endif
@@ -29,6 +31,8 @@
     return lock;
   }
   return new RWLockGeneric();
+#elif defined(WEBRTC_ANDROID)
+  return new RWLockGeneric();
 #else
   return RWLockPosix::Create();
 #endif
