Index: webrtc/sources/system_wrappers/source/Android.mk
===================================================================
--- webrtc.orig/sources/system_wrappers/source/Android.mk	2012-06-25 11:16:56.000000000 +0200
+++ webrtc/sources/system_wrappers/source/Android.mk	2012-06-25 11:37:27.051155227 +0200
@@ -20,6 +20,7 @@
     android/cpu-features.c \
     cpu_features_android.c \
     map.cc \
+    rw_lock_generic.cc \
     sort.cc \
     aligned_malloc.cc \
     atomic32_posix.cc \
Index: webrtc/sources/system_wrappers/source/rw_lock.cc
===================================================================
--- webrtc.orig/sources/system_wrappers/source/rw_lock.cc	2012-06-25 11:16:01.083209010 +0200
+++ webrtc/sources/system_wrappers/source/rw_lock.cc	2012-06-25 11:37:27.051155227 +0200
@@ -14,6 +14,9 @@
 
 #if defined(_WIN32)
     #include "rw_lock_win.h"
+#elif defined(WEBRTC_ANDROID)
+	#include <stdlib.h>
+	#include "rw_lock_generic.h"
 #else
     #include "rw_lock_posix.h"
 #endif
@@ -23,6 +26,8 @@
 {
 #ifdef _WIN32
     RWLockWrapper* lock =  new RWLockWindows();
+#elif defined(WEBRTC_ANDROID)
+    RWLockWrapper* lock =  new RWLockWrapperGeneric();
 #else
     RWLockWrapper* lock =  new RWLockPosix();
 #endif
