Index: webrtc/sources/system_wrappers/source/Android.mk
===================================================================
--- webrtc.orig/sources/system_wrappers/source/Android.mk	2012-05-05 12:54:35.398284261 +0200
+++ webrtc/sources/system_wrappers/source/Android.mk	2012-05-05 12:54:58.306284748 +0200
@@ -18,6 +18,7 @@
 LOCAL_CPP_EXTENSION := .cc
 LOCAL_SRC_FILES := \
     map.cc \
+    rw_lock_generic.cc \
     sort.cc \
     aligned_malloc.cc \
     atomic32_posix.cc \
Index: webrtc/sources/system_wrappers/source/rw_lock.cc
===================================================================
--- webrtc.orig/sources/system_wrappers/source/rw_lock.cc	2012-05-05 12:54:23.326284004 +0200
+++ webrtc/sources/system_wrappers/source/rw_lock.cc	2012-05-05 12:54:58.306284748 +0200
@@ -14,6 +14,9 @@
 
 #if defined(_WIN32)
     #include "rw_lock_win.h"
+#elif defined(WEBRTC_ANDROID)
+	#include <stdlib.h>
+	#include "rw_lock_generic.h"
 #else
     #include "rw_lock_posix.h"
 #endif
@@ -23,6 +26,8 @@
 {
 #ifdef _WIN32
     RWLockWrapper* lock =  new RWLockWindows();
+#elif defined(WEBRTC_ANDROID)
+    RWLockWrapper* lock =  new RWLockWrapperGeneric();
 #else
     RWLockWrapper* lock =  new RWLockPosix();
 #endif
