Index: webrtc/sources/modules/audio_coding/main/source/Android.mk
===================================================================
--- webrtc.orig/sources/modules/audio_coding/main/source/Android.mk	2012-09-26 22:52:45.000000000 +0200
+++ webrtc/sources/modules/audio_coding/main/source/Android.mk	2012-09-26 22:59:20.666745998 +0200
@@ -45,8 +45,8 @@
     $(LOCAL_PATH)/../../codecs/g711/include \
     $(LOCAL_PATH)/../../codecs/g722/include \
     $(LOCAL_PATH)/../../codecs/ilbc/interface \
-    $(LOCAL_PATH)/../../codecs/iSAC/main/interface \
-    $(LOCAL_PATH)/../../codecs/iSAC/fix/interface \
+    $(LOCAL_PATH)/../../codecs/isac/main/interface \
+    $(LOCAL_PATH)/../../codecs/isac/fix/interface \
     $(LOCAL_PATH)/../../codecs/pcm16b/include \
     $(LOCAL_PATH)/../../neteq/interface \
     $(LOCAL_PATH)/../../../.. \
Index: webrtc/sources/modules/audio_coding/codecs/isac/fix/source/Android.mk
===================================================================
--- webrtc.orig/sources/modules/audio_coding/codecs/isac/fix/source/Android.mk	2012-09-26 22:53:34.000000000 +0200
+++ webrtc/sources/modules/audio_coding/codecs/isac/fix/source/Android.mk	2012-09-26 23:01:13.990741258 +0200
@@ -52,7 +52,8 @@
     pitch_filter_armv6.S
 else
 LOCAL_SRC_FILES += \
-    lattice_c.c
+    lattice_c.c \
+    pitch_filter_c.c
 endif
 
 # Flags passed to both C and C++ files.
Index: webrtc/sources/common_video/libyuv/Android.mk
===================================================================
--- webrtc.orig/sources/common_video/libyuv/Android.mk	2012-09-26 22:59:07.000000000 +0200
+++ webrtc/sources/common_video/libyuv/Android.mk	2012-09-26 22:59:20.670745997 +0200
@@ -17,7 +17,7 @@
 LOCAL_MODULE_TAGS := optional
 LOCAL_CPP_EXTENSION := .cc
 LOCAL_SRC_FILES := \
-    libyuv.cc \
+    webrtc_libyuv.cc \
     scaler.cc
 
 # Flags passed to both C and C++ files.
