Index: webrtc/sources/modules/audio_coding/main/source/Android.mk
===================================================================
--- webrtc.orig/sources/modules/audio_coding/main/source/Android.mk	2011-11-24 18:22:13.000000000 +0100
+++ webrtc/sources/modules/audio_coding/main/source/Android.mk	2012-08-15 16:14:18.217830820 +0200
@@ -45,8 +45,8 @@
     $(LOCAL_PATH)/../../codecs/g711/include \
     $(LOCAL_PATH)/../../codecs/g722/include \
     $(LOCAL_PATH)/../../codecs/ilbc/interface \
-    $(LOCAL_PATH)/../../codecs/iSAC/main/interface \
-    $(LOCAL_PATH)/../../codecs/iSAC/fix/interface \
+    $(LOCAL_PATH)/../../codecs/isac/main/interface \
+    $(LOCAL_PATH)/../../codecs/isac/fix/interface \
     $(LOCAL_PATH)/../../codecs/pcm16b/include \
     $(LOCAL_PATH)/../../neteq/interface \
     $(LOCAL_PATH)/../../../.. \
Index: webrtc/sources/modules/audio_coding/codecs/isac/fix/source/Android.mk
===================================================================
--- webrtc.orig/sources/modules/audio_coding/codecs/isac/fix/source/Android.mk	2012-08-15 15:57:37.329872683 +0200
+++ webrtc/sources/modules/audio_coding/codecs/isac/fix/source/Android.mk	2012-08-15 16:18:20.633820683 +0200
@@ -52,7 +52,8 @@
     pitchfilter_armv6.S
 else
 LOCAL_SRC_FILES += \
-    lattice_c.c
+    lattice_c.c \
+    pitch_filter_c.c
 endif
 
 # Flags passed to both C and C++ files.
