Index: webrtc/sources/modules/audio_processing/ns/Android.mk
===================================================================
--- webrtc.orig/sources/modules/audio_processing/ns/Android.mk	2012-09-08 12:30:36.523299438 +0200
+++ webrtc/sources/modules/audio_processing/ns/Android.mk	2012-09-08 14:35:43.573724486 +0200
@@ -70,7 +70,7 @@
             $(TARGET_C_INCLUDES)) -S -o $@ $^
 
 LOCAL_GENERATED_SOURCES := $(GEN)
-LOCAL_SRC_FILES := nsx_core_neon.S
+LOCAL_SRC_FILES := nsx_core_neon.c
 
 # Flags passed to both C and C++ files.
 LOCAL_CFLAGS := \
