Index: webrtc/sources/modules/audio_processing/ns/Android.mk
===================================================================
--- webrtc.orig/sources/modules/audio_processing/ns/Android.mk	2012-09-28 20:47:19.794686927 +0200
+++ webrtc/sources/modules/audio_processing/ns/Android.mk	2012-09-28 20:47:20.823200883 +0200
@@ -70,7 +70,7 @@
             $(TARGET_C_INCLUDES)) -S -o $@ $^
 
 LOCAL_GENERATED_SOURCES := $(GEN)
-LOCAL_SRC_FILES := nsx_core_neon.S
+LOCAL_SRC_FILES := nsx_core_neon.c
 
 # Flags passed to both C and C++ files.
 LOCAL_CFLAGS := \
Index: webrtc/sources/modules/audio_processing/aecm/Android.mk
===================================================================
--- webrtc.orig/sources/modules/audio_processing/aecm/Android.mk	2012-09-28 20:47:19.794686927 +0200
+++ webrtc/sources/modules/audio_processing/aecm/Android.mk	2012-09-28 20:48:31.078307941 +0200
@@ -70,7 +70,7 @@
             $(TARGET_C_INCLUDES)) -S -o $@ $^
 
 LOCAL_GENERATED_SOURCES := $(GEN)
-LOCAL_SRC_FILES := aecm_core_neon.S
+LOCAL_SRC_FILES := aecm_core_neon.c
 
 # Flags passed to both C and C++ files.
 LOCAL_CFLAGS := \
