Index: ffmpeg/ffmpeg_src/libavutil/arm/intmath.h
===================================================================
--- ffmpeg.orig/ffmpeg_src/libavutil/arm/intmath.h	2012-11-13 23:36:46.000000000 +0100
+++ ffmpeg/ffmpeg_src/libavutil/arm/intmath.h	2012-11-13 23:42:25.009158265 +0100
@@ -75,7 +75,7 @@
     return x;
 }
 
-#if !CONFIG_SMALL //the code below cannot be compiled without always_inline
+#if !CONFIG_SMALL && !__ANDROID__ //the code below cannot be compiled without always_inline
 #define av_clip_uintp2 av_clip_uintp2_arm
 static av_always_inline av_const unsigned av_clip_uintp2_arm(int a, int p)
 {
