Index: opus/sources/src/opus_decoder.c
===================================================================
--- opus.orig/sources/src/opus_decoder.c	2012-05-12 19:32:29.379114960 +0200
+++ opus/sources/src/opus_decoder.c	2012-05-12 19:33:26.455112569 +0200
@@ -454,7 +454,8 @@
    {
       const CELTMode *celt_mode;
       celt_decoder_ctl(celt_dec, CELT_GET_MODE(&celt_mode));
-      window = celt_mode->window;
+      if(celt_mode != NULL)
+    	  window = celt_mode->window;
    }
 
    /* 5 ms redundant frame for SILK->CELT */
