Index: pjsip/sources/pjlib/include/pj/compat/os_auto.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ pjsip/sources/pjlib/include/pj/compat/os_auto.h	2013-03-28 22:03:12.580814663 +0100
@@ -0,0 +1,136 @@
+/* $Id: os_linux.h 3553 2011-05-05 06:14:19Z nanang $ */
+/* 
+ * Copyright (C) 2008-2011 Teluu Inc. (http://www.teluu.com)
+ * Copyright (C) 2003-2008 Benny Prijono <benny@prijono.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA 
+ */
+#ifndef __PJ_COMPAT_OS_AUTO_H__
+#define __PJ_COMPAT_OS_AUTO_H__
+
+/**
+ * @file os_linux.h
+ * @brief Describes Linux operating system specifics.
+ */
+
+#define PJ_OS_NAME		    "android"
+
+#define PJ_HAS_ARPA_INET_H	    1
+#define PJ_HAS_ASSERT_H		    1
+#define PJ_HAS_CTYPE_H		    1
+#define PJ_HAS_ERRNO_H		    1
+#define PJ_HAS_LINUX_SOCKET_H	    0
+#define PJ_HAS_MALLOC_H		    1
+#define PJ_HAS_NETDB_H		    1
+#define PJ_HAS_NETINET_IN_H	    1
+#define PJ_HAS_NETINET_TCP_H    1
+#define PJ_HAS_SETJMP_H		    1
+#define PJ_HAS_STDARG_H		    1
+#define PJ_HAS_STDDEF_H		    1
+#define PJ_HAS_STDIO_H		    1
+#define PJ_HAS_STDLIB_H		    1
+#define PJ_HAS_STDINT_H         1
+#define PJ_HAS_STRING_H		    1
+#define PJ_HAS_SYS_IOCTL_H	    1
+#define PJ_HAS_SYS_SELECT_H	    1
+#define PJ_HAS_SYS_SOCKET_H	    1
+#define PJ_HAS_SYS_TIME_H	    1
+#define PJ_HAS_SYS_TIMEB_H	    1
+#define PJ_HAS_SYS_TYPES_H	    1
+#define PJ_HAS_TIME_H		    1
+#define PJ_HAS_UNISTD_H		    1
+#define PJ_HAS_SEMAPHORE_H	    1
+
+#define PJ_HAS_MSWSOCK_H	    0
+#define PJ_HAS_WINSOCK_H	    0
+#define PJ_HAS_WINSOCK2_H	    0
+
+#define PJ_SOCK_HAS_INET_ATON	    1
+#define PJ_SOCK_HAS_INET_NTOP	    1
+#define PJ_SOCK_HAS_INET_PTON	1
+
+/* Set 1 if native sockaddr_in has sin_len member. 
+ * Default: 0
+ */
+#define PJ_SOCKADDR_HAS_LEN	    0
+
+/**
+ * If this macro is set, it tells select I/O Queue that select() needs to
+ * be given correct value of nfds (i.e. largest fd + 1). This requires
+ * select ioqueue to re-scan the descriptors on each registration and
+ * unregistration.
+ * If this macro is not set, then ioqueue will always give FD_SETSIZE for
+ * nfds argument when calling select().
+ *
+ * Default: 0
+ */
+#define PJ_SELECT_NEEDS_NFDS	    1
+
+/* Is errno a good way to retrieve OS errors?
+ */
+#define PJ_HAS_ERRNO_VAR	    1
+
+/* When this macro is set, getsockopt(SOL_SOCKET, SO_ERROR) will return
+ * the status of non-blocking connect() operation.
+ */
+#define PJ_HAS_SO_ERROR             1
+
+/* This value specifies the value set in errno by the OS when a non-blocking
+ * socket recv() can not return immediate daata.
+ */
+#define PJ_BLOCKING_ERROR_VAL       EAGAIN
+
+/* This value specifies the value set in errno by the OS when a non-blocking
+ * socket connect() can not get connected immediately.
+ */
+#define PJ_BLOCKING_CONNECT_ERROR_VAL   EINPROGRESS
+
+/* Default threading is enabled, unless it's overridden. */
+#ifndef PJ_HAS_THREADS
+#  define PJ_HAS_THREADS	    (1)
+#endif
+
+#define PJ_HAS_HIGH_RES_TIMER	    1
+#define PJ_HAS_MALLOC               1
+#ifndef PJ_OS_HAS_CHECK_STACK
+#   define PJ_OS_HAS_CHECK_STACK    0
+#endif
+#define PJ_NATIVE_STRING_IS_UNICODE 0
+
+#define PJ_ATOMIC_VALUE_TYPE	    long
+
+/* 
+ * pthread_rwlock_t does not appear to be defined on all Android arch
+ * From android 4 it seems to be there, so enable it
+ */
+#define PJ_EMULATE_RWMUTEX	    0
+
+/* If 1, pj_thread_create() should enforce the stack size when creating 
+ * threads.
+ * Default: 0 (let OS decide the thread's stack size).
+ */
+#define PJ_THREAD_SET_STACK_SIZE    	0
+
+/* If 1, pj_thread_create() should allocate stack from the pool supplied.
+ * Default: 0 (let OS allocate memory for thread's stack).
+ */
+#define PJ_THREAD_ALLOCATE_STACK    	0
+
+/* Linux has socklen_t */
+#define PJ_HAS_SOCKLEN_T		1
+
+
+#endif	/* __PJ_COMPAT_OS_AUTO_H__ */
+
Index: pjsip/sources/pjlib/include/pj/config_site.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ pjsip/sources/pjlib/include/pj/config_site.h	2013-03-28 21:54:16.848837068 +0100
@@ -0,0 +1,134 @@
+//We distribute only for android so by default activate it
+#ifndef PJ_ANDROID
+#   define PJ_ANDROID   1
+#endif
+
+#define PJMEDIA_SDP_NEG_PREFER_REMOTE_CODEC_ORDER 0
+#define PJ_HAS_IPV6 1
+
+#define PJ_STUN_KEEP_ALIVE_SEC 90
+#define PJ_ICE_SESS_KEEP_ALIVE_MIN 90
+
+/* Fix google voice problem with RTP DTMF events */
+#define PJMEDIA_RTP_PT_TELEPHONE_EVENTS          101
+#define PJMEDIA_RTP_PT_TELEPHONE_EVENTS_STR      "101"
+
+#ifndef USE_CSIPSIMPLE_HACKS
+#	define USE_CSIPSIMPLE_HACKS 1
+#endif
+
+#define PJSUA_HAS_VIDEO 1
+#define PJSIP_AUTH_HEADER_CACHING   1
+
+// CSipSimple hacks
+#if defined(USE_CSIPSIMPLE_HACKS) && USE_CSIPSIMPLE_HACKS!=0
+#ifdef __cplusplus
+extern "C" {
+#endif
+extern int css_tcp_keep_alive_interval;
+extern int css_tls_keep_alive_interval;
+#ifdef __cplusplus
+}
+#endif
+#	define PJSIP_TCP_KEEP_ALIVE_INTERVAL css_tcp_keep_alive_interval
+#	define PJSIP_TLS_KEEP_ALIVE_INTERVAL css_tls_keep_alive_interval
+
+#endif
+
+
+#if defined(PJ_ANDROID) && PJ_ANDROID!=0
+
+	#define PJSIP_TCP_TRANSPORT_BACKLOG 50
+
+//    #define PJ_LOG_USE_STACK_BUFFER             1
+	//#define PJSUA_DEFAULT_AUDIO_FRAME_PTIME 30
+
+	/* Optimizations */
+//	#define PJMEDIA_HAS_ALAW_ULAW_TABLE 1
+	#define PJ_ENABLE_EXTRA_CHECK   0
+
+	//Conference switch board
+	#define PJMEDIA_CONF_USE_SWITCH_BOARD 0
+
+    #define PJ_OS_HAS_CHECK_STACK		0
+	#define PJ_QOS_IMPLEMENTATION    PJ_QOS_BSD
+
+
+	#define PJSIP_CHECK_VIA_SENT_BY 0
+
+	/*
+     * PJMEDIA settings
+     */
+
+    /* Disable non-Android audio devices */
+    #define PJMEDIA_AUDIO_DEV_HAS_PORTAUDIO     0
+    #define PJMEDIA_AUDIO_DEV_HAS_WMME          0
+    #define PJMEDIA_AUDIO_DEV_HAS_ANDROID       1
+
+    /* Select codecs to disable */
+    #define PJMEDIA_HAS_L16_CODEC               0
+
+    /* Fine tune Speex's default settings for best performance/quality */
+    #define PJMEDIA_CODEC_SPEEX_DEFAULT_QUALITY 5
+
+    /* If no floating point, no speex AEC and use the echo suppressor. */
+    #define PJMEDIA_HAS_SPEEX_AEC		PJ_HAS_FLOATING_POINT
+	/* If no floating point, no iLBC codec. */
+#if defined(PJMEDIA_HAS_ILBC_CODEC) && PJMEDIA_HAS_ILBC_CODEC == 1
+	#undef PJMEDIA_HAS_ILBC_CODEC
+	#define PJMEDIA_HAS_ILBC_CODEC 		PJ_HAS_FLOATING_POINT
+#endif
+
+    /* Previously, resampling is disabled due to performance reason and
+     * this condition prevented some 'light' wideband codecs (e.g: G722.1)
+     * to work along with narrowband codecs. Lately, some tests showed
+     * that 16kHz <-> 8kHz resampling using libresample small filter was
+     * affordable on ARM9 222 MHz, so here we decided to enable resampling.
+     * Note that it is important to make sure that libresample is created
+     * using small filter. For example PJSUA_DEFAULT_CODEC_QUALITY must
+     * be set to 3 or 4 so pjsua-lib will apply small filter resampling.
+     */
+    //#define PJMEDIA_RESAMPLE_IMP              PJMEDIA_RESAMPLE_NONE
+    #define PJMEDIA_RESAMPLE_IMP                PJMEDIA_RESAMPLE_LIBRESAMPLE
+
+    /* Use the lighter WSOLA implementation */
+    #define PJMEDIA_WSOLA_IMP                   PJMEDIA_WSOLA_IMP_WSOLA_LITE
+
+
+//	#define PJMEDIA_SND_DEFAULT_REC_LATENCY 150
+//	#define PJMEDIA_SND_DEFAULT_PLAY_LATENCY 200
+
+    /*
+     * PJSUA settings.
+     */
+
+    /* Default codec quality, previously was set to 5, however it is now
+     * set to 4 to make sure pjsua instantiates resampler with small filter.
+     */
+    #define PJSUA_DEFAULT_CODEC_QUALITY         4
+	/**
+	 * We need hi availablitly so retry each minute if network fails -
+	 * all the more so as we unregister if network not available for us
+	 */
+	#define PJSUA_REG_RETRY_INTERVAL	60
+	//#define PJSUA_DEFAULT_ILBC_MODE 			20
+
+
+    /* Set maximum number of dialog/transaction/calls to minimum */
+    #define PJSIP_MAX_TSX_COUNT 		31
+    #define PJSIP_MAX_DIALOG_COUNT 		31
+    #define PJSUA_MAX_CALLS			4
+
+	/* Other pjsua settings */
+        /* We have to care about local accounts. So upscale it */
+	#define PJSUA_MAX_ACC			20
+        #define PJSUA_MAX_PLAYERS			4
+	#define PJSUA_MAX_RECORDERS			4
+	#define PJSUA_MAX_CONF_PORTS		(PJSUA_MAX_CALLS+2*PJSUA_MAX_PLAYERS)
+	#define PJSUA_MAX_BUDDIES			32
+
+	/* Seems to fix problems when restarting too frequently */
+	#define PJ_IOQUEUE_MAX_HANDLES	    1024
+
+
+#endif
Index: pjsip/sources/pjmedia/include/pjmedia/config_auto.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ pjsip/sources/pjmedia/include/pjmedia/config_auto.h	2013-03-28 21:54:16.848837068 +0100
@@ -0,0 +1,38 @@
+/* $Id: config_auto.h.in 3295 2010-08-25 12:51:29Z bennylp $ */
+/*
+ * Copyright (C) 2008-2009 Teluu Inc. (http://www.teluu.com)
+ * Copyright (C) 2003-2008 Benny Prijono <benny@prijono.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#ifndef __PJMEDIA_CONFIG_AUTO_H_
+#define __PJMEDIA_CONFIG_AUTO_H_
+
+/**
+ * @file config_auto.h
+ * @brief PJMEDIA configuration as set by autoconf script
+ */
+
+/*
+ * Note:
+ *	The configuration in config_site.h overrides any other settings,
+ *	including the setting as detected by autoconf.
+ */
+
+
+#endif	/* __PJMEDIA_CONFIG_AUTO_H_ */
+
+
+
Index: pjsip/sources/pjsip/include/pjsip/sip_autoconf.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ pjsip/sources/pjsip/include/pjsip/sip_autoconf.h	2013-03-28 21:54:16.848837068 +0100
@@ -0,0 +1,30 @@
+/* $Id: sip_autoconf.h.in 3295 2010-08-25 12:51:29Z bennylp $ */
+/*
+ * Copyright (C) 2008-2009 Teluu Inc. (http://www.teluu.com)
+ * Copyright (C) 2003-2008 Benny Prijono <benny@prijono.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#ifndef __PJSIP_SIP_AUTOCONF_H__
+#define __PJSIP_SIP_AUTOCONF_H__
+
+/**
+ * @file sip_autoconf.h
+ * @brief Describes operating system specifics (automatically detected by
+ *        autoconf)
+ */
+
+#endif	/* __PJSIP_SIP_AUTOCONF_H__ */
+
Index: pjsip/sources/pjlib/include/pj/compat/m_auto.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ pjsip/sources/pjlib/include/pj/compat/m_auto.h	2013-03-28 21:54:16.848837068 +0100
@@ -0,0 +1,38 @@
+/* $Id: m_auto.h.in 3295 2010-08-25 12:51:29Z bennylp $ */
+/*
+ * Copyright (C) 2008-2009 Teluu Inc. (http://www.teluu.com)
+ * Copyright (C) 2003-2008 Benny Prijono <benny@prijono.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#ifndef __PJ_COMPAT_M_AUTO_H__
+#define __PJ_COMPAT_M_AUTO_H__
+
+/**
+ * @file m_auto.h
+ * @brief Automatically generated process definition file.
+ */
+
+#define PJ_M_NAME "android"
+
+
+#define PJ_IS_LITTLE_ENDIAN 1
+#define PJ_IS_BIG_ENDIAN 0
+
+/* Deprecated */
+#define PJ_HAS_PENTIUM		0
+
+#endif	/* __PJ_COMPAT_M_AUTO_H__ */
+
Index: pjsip/sources/pjmedia/include/pjmedia-codec/config_auto.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ pjsip/sources/pjmedia/include/pjmedia-codec/config_auto.h	2013-03-28 21:54:16.848837068 +0100
@@ -0,0 +1,32 @@
+/* $Id: config_auto.h.in 3841 2011-10-24 09:28:13Z ming $ */
+/*
+ * Copyright (C) 2008-2009 Teluu Inc. (http://www.teluu.com)
+ * Copyright (C) 2003-2008 Benny Prijono <benny@prijono.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#ifndef __PJMEDIA_CODEC_CONFIG_AUTO_H_
+#define __PJMEDIA_CODEC_CONFIG_AUTO_H_
+
+/**
+ * @file config_auto.h
+ * @brief PJMEDIA-CODEC configuration as set by autoconf script
+ */
+
+
+
+#endif	/* __PJMEDIA_CODEC_CONFIG_AUTO_H_ */
+
+
