Index: pjsip/sources/pjmedia/include/pjmedia-codec/types.h
===================================================================
--- pjsip.orig/sources/pjmedia/include/pjmedia-codec/types.h	2011-11-21 08:52:39.865661432 +0100
+++ pjsip/sources/pjmedia/include/pjmedia-codec/types.h	2011-11-22 22:23:20.174183882 +0100
@@ -60,9 +60,17 @@
     PJMEDIA_RTP_PT_START = (PJMEDIA_RTP_PT_DYNAMIC-1),
 #endif
 
+    PJMEDIA_RTP_PT_ILBC,			/**< iLBC (13.3/15.2Kbps)   */ // r3gis3r : changed position cause 102 in webRTC is iLBC
+
+#if defined(PJMEDIA_HAS_WEBRTC_CODEC) && PJMEDIA_HAS_WEBRTC_CODEC != 0
+    PJMEDIA_RTP_PT_ISAC_WB,
+    PJMEDIA_RTP_PT_ISAC_UWB,
+#endif
+
     PJMEDIA_RTP_PT_SPEEX_NB,			/**< Speex narrowband/8KHz  */
     PJMEDIA_RTP_PT_SPEEX_WB,			/**< Speex wideband/16KHz   */
     PJMEDIA_RTP_PT_SPEEX_UWB,			/**< Speex 32KHz	    */
+#if defined(PJMEDIA_HAS_L16_CODEC) && PJMEDIA_HAS_L16_CODEC != 0
     PJMEDIA_RTP_PT_L16_8KHZ_MONO,		/**< L16 @ 8KHz, mono	    */
     PJMEDIA_RTP_PT_L16_8KHZ_STEREO,		/**< L16 @ 8KHz, stereo     */
     //PJMEDIA_RTP_PT_L16_11KHZ_MONO,		/**< L16 @ 11KHz, mono	    */
@@ -75,14 +83,26 @@
     //PJMEDIA_RTP_PT_L16_32KHZ_STEREO,		/**< L16 @ 32KHz, stereo    */
     //PJMEDIA_RTP_PT_L16_48KHZ_MONO,		/**< L16 @ 48KHz, mono	    */
     //PJMEDIA_RTP_PT_L16_48KHZ_STEREO,		/**< L16 @ 48KHz, stereo    */
-    PJMEDIA_RTP_PT_ILBC,			/**< iLBC (13.3/15.2Kbps)   */
+#endif
+
+#if ( defined(PJMEDIA_HAS_INTEL_IPP) && PJMEDIA_HAS_INTEL_IPP != 0 ) \
+	|| ( defined(PJMEDIA_HAS_PASSTHROUGH_CODECS) && PJMEDIA_HAS_PASSTHROUGH_CODECS!=0 ) \
+	|| ( defined(PJMEDIA_HAS_AMR_STAGEFRIGHT_CODEC) && PJMEDIA_HAS_AMR_STAGEFRIGHT_CODEC!=0 )
     PJMEDIA_RTP_PT_AMR,				/**< AMR (4.75 - 12.2Kbps)  */
     PJMEDIA_RTP_PT_AMRWB,			/**< AMRWB (6.6 - 23.85Kbps)*/
     PJMEDIA_RTP_PT_AMRWBE,			/**< AMRWBE		    */
+#endif
+
+#if ( defined(PJMEDIA_HAS_INTEL_IPP) && PJMEDIA_HAS_INTEL_IPP != 0 ) \
+		|| ( defined(PJMEDIA_HAS_G726_CODEC) && PJMEDIA_HAS_G726_CODEC!=0 )
     PJMEDIA_RTP_PT_G726_16,			/**< G726 @ 16Kbps	    */
     PJMEDIA_RTP_PT_G726_24,			/**< G726 @ 24Kbps	    */
     PJMEDIA_RTP_PT_G726_32,			/**< G726 @ 32Kbps	    */
     PJMEDIA_RTP_PT_G726_40,			/**< G726 @ 40Kbps	    */
+#endif
+
+#if ( defined(PJMEDIA_HAS_G7221_CODEC) && PJMEDIA_HAS_G7221_CODEC!=0 ) \
+	|| ( defined(PJMEDIA_HAS_INTEL_IPP) && PJMEDIA_HAS_INTEL_IPP != 0 )
     PJMEDIA_RTP_PT_G722_1_16,			/**< G722.1 (16Kbps)	    */
     PJMEDIA_RTP_PT_G722_1_24,			/**< G722.1 (24Kbps)	    */
     PJMEDIA_RTP_PT_G722_1_32,			/**< G722.1 (32Kbps)	    */
@@ -91,7 +111,17 @@
     PJMEDIA_RTP_PT_G7221C_48,			/**< G722.1 Annex C (48Kbps)*/
     PJMEDIA_RTP_PT_G7221_RSV1,			/**< G722.1 reserve	    */
     PJMEDIA_RTP_PT_G7221_RSV2,			/**< G722.1 reserve	    */
+#endif
 
+#if ( defined(PJMEDIA_HAS_SILK_CODEC) && PJMEDIA_HAS_SILK_CODEC!=0 )
+    PJMEDIA_RTP_PT_SILK_NB,				/**< SILK @ 8KHz	*/
+    PJMEDIA_RTP_PT_SILK_WB,				/**< SILK @ 16KHz	*/
+    PJMEDIA_RTP_PT_SILK_UWB,			/**< SILK @ 24KHz	*/
+#endif
+
+#if ( defined(PJMEDIA_HAS_CODEC2_CODEC) && PJMEDIA_HAS_CODEC2_CODEC!=0 )
+    PJMEDIA_RTP_PT_CODEC2,
+#endif
     /* Caution!
      * Ensure the value of the last pt above is <= 127.
      */
