Index: pjsip/sources/pjsip/src/pjsua-lib/pjsua_acc.c
===================================================================
--- pjsip.orig/sources/pjsip/src/pjsua-lib/pjsua_acc.c	2012-04-25 14:54:17.202673575 +0200
+++ pjsip/sources/pjsip/src/pjsua-lib/pjsua_acc.c	2012-04-25 14:54:25.638673755 +0200
@@ -2603,6 +2603,11 @@
 
 
     /* Create the contact header */
+	pj_str_t user_tmp;
+	user_tmp.ptr = (char*)pj_pool_alloc(pool, PJSIP_MAX_URL_SIZE);
+    const pjsip_parser_const_t *pc = pjsip_parser_const();
+	pj_strncpy_escape(&user_tmp, &acc->user_part, PJSIP_MAX_URL_SIZE, &pc->pjsip_USER_SPEC);
+
     contact->ptr = (char*)pj_pool_alloc(pool, PJSIP_MAX_URL_SIZE);
     contact->slen = pj_ansi_snprintf(contact->ptr, PJSIP_MAX_URL_SIZE,
 				     "%s%.*s%s<%s:%.*s%s%s%.*s%s:%d%s%.*s%s>%.*s",
@@ -2611,8 +2616,8 @@
 				     acc->display.ptr,
 				     (acc->display.slen?"\" " : ""),
 				     (secure ? PJSUA_SECURE_SCHEME : "sip"),
-				     (int)acc->user_part.slen,
-				     acc->user_part.ptr,
+				     (int)user_tmp.slen,
+				     user_tmp.ptr,
 				     (acc->user_part.slen?"@":""),
 				     beginquote,
 				     (int)local_addr.slen,
