Index: pjsip/sources/pjsip/src/pjsua-lib/pjsua_call.c
===================================================================
--- pjsip.orig/sources/pjsip/src/pjsua-lib/pjsua_call.c	2013-03-28 21:43:50.404863268 +0100
+++ pjsip/sources/pjsip/src/pjsua-lib/pjsua_call.c	2013-03-28 21:44:16.668862170 +0100
@@ -23,6 +23,9 @@
 
 #define THIS_FILE		"pjsua_call.c"
 
+#if USE_CSIPSIMPLE_HACKS
+	pj_bool_t pjsua_no_update = PJ_FALSE;
+#endif
 
 /* Retry interval of sending re-INVITE for locking a codec when remote
  * SDP answer contains multiple codec, in milliseconds.
@@ -3341,7 +3344,11 @@
     }
 
     
-    if (rem_can_update) {
+    if (rem_can_update
+#if USE_CSIPSIMPLE_HACKS
+            && !pjsua_no_update
+#endif
+    ) {
 	status = pjsip_inv_update(inv, NULL, new_offer, &tdata);
     } else {
 	status = pjsip_inv_reinvite(inv, NULL, new_offer, &tdata);
