Index: pjsip/sources/pjmedia/src/pjmedia-codec/ffmpeg_codecs.c
===================================================================
--- pjsip.orig/sources/pjmedia/src/pjmedia-codec/ffmpeg_codecs.c	2012-01-08 16:08:36.439047748 +0100
+++ pjsip/sources/pjmedia/src/pjmedia-codec/ffmpeg_codecs.c	2012-01-08 20:51:05.995096055 +0100
@@ -918,7 +918,8 @@
     /* Decoded format */
     pjmedia_format_init_video(&attr->dec_fmt, desc->info.dec_fmt_id[0],
                               //352, 288, 30000, 1001);
-                              720, 576, 30000, 1001);
+							  720, 480, 30000, 1001);
+                              //720, 576, 30000, 1001);
 
     /* Decoding fmtp */
     attr->dec_fmtp = desc->dec_fmtp;
