Index: pjsip/sources/pjsip/src/pjsua-lib/pjsua_media.c
===================================================================
--- pjsip.orig/sources/pjsip/src/pjsua-lib/pjsua_media.c	2012-05-18 22:17:41.688514281 +0200
+++ pjsip/sources/pjsip/src/pjsua-lib/pjsua_media.c	2012-05-18 22:29:32.756484541 +0200
@@ -1403,6 +1403,15 @@
 	    pjsua_set_media_tp_state(call_med, PJSUA_MED_TP_NULL);
 	    pjmedia_transport_close(call_med->tp);
 	    call_med->tp = call_med->tp_orig = NULL;
+
+		if (call_med->type == PJMEDIA_TYPE_AUDIO) {
+		    pjsua_aud_stop_stream(call_med);
+		}
+#if PJMEDIA_HAS_VIDEO
+		else if (call_med->type == PJMEDIA_TYPE_VIDEO) {
+			pjsua_vid_stop_stream(call_med);
+		}
+#endif
 	}
     }
 }
