Index: pjsip/sources/pjlib/src/pj/os_core_unix.c
===================================================================
--- pjsip.orig/sources/pjlib/src/pj/os_core_unix.c	2015-02-08 15:54:42.791139902 +0100
+++ pjsip/sources/pjlib/src/pj/os_core_unix.c	2015-02-08 15:54:42.707132287 +0100
@@ -716,6 +716,9 @@
     int result;
 
     PJ_CHECK_STACK();
+    if(pj_thread_this()->thread == rec->thread){
+    	return PJ_SUCCESS;
+    }
 
     PJ_LOG(6, (pj_thread_this()->obj_name, "Joining thread %s", p->obj_name));
     result = pthread_join( rec->thread, &ret);
